src/components/AssetIcon.tsx

```diff
@@ -79,31 +82,48 @@ const AssetWithNetwork: React.FC<AssetWithNetworkProps> = ({
   // Failure to check this means we would lose loading FOX icon functionality
   const showFallback = !asset.icon && !asset.icons?.length
 
+  const [imageLoaded, setImageLoaded] = useState(false)
+  const [networkImageLoaded, setNetworkImageLoaded] = useState(!showNetwork)
+
+  const handleImageLoaded = useCallback(() => setImageLoaded(true), [])
+  const handleNetworkImageLoaded = useCallback(() => setNetworkImageLoaded(true), [])
+
+  const areImagesLoaded = useMemo(
+    () => (imageLoaded && networkImageLoaded) || !shouldLazyLoad,
+    [imageLoaded, networkImageLoaded, shouldLazyLoad],
+  )
```

Comment:
suggestion (cleanup):

While the logic is effectively the same as `<LazyLoadAvatar />`, feels like this is now adding additional logic (to handle lazy-loading)/branching (to handle `shouldLazyLoad`) to this component for the sake of lazy-loading.

Wondering if we could either:

- Make lazy the default here to avoid branching
- Wrap `<AssetIcon />` into a similar `<LazyLoadAssetIcon />`

Reply:
🧠 

done in [acd3eca](https://github.com/shapeshift/web/pull/7492/commits/acd3eca4e874509c8e7c46370c89c657228d614e)